### PR TITLE
MULE-16725: Fix race condition in flow lifecycle handling

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java [hunk 1]
```diff
@@ -58,8 +58,8 @@
 
   private ApplicationContext parentContext;
   private MuleArtifactContext muleArtifactContext;
-  private ArtifactType artifactType;
-  private ComponentBuildingDefinitionProvider runtimeComponentBuildingDefinitionsProvider;
+  private final ArtifactType artifactType;
+  private final ComponentBuildingDefinitionProvider runtimeComponentBuildingDefinitionsProvider;
```
Comment (plbalbi) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java]: For this two fields that are changed to a **final** modifier, isn't ir required to set them in the static constructor?

File: core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java [hunk 2]
```diff
@@ -52,31 +53,31 @@
   /**
    * A reference to Mule's internal registry
    */
-  private Registry registry;
+  private final Registry registry;
```
Comment (plbalbi) [inline core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java]: Also this one?

File: core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java [hunk 3]
```diff
@@ -52,31 +53,31 @@
...
 
   /**
    * We cache transformer searches so that we only search once
    */
   protected Map<String, Transformer> exactTransformerCache = new ConcurrentHashMap<>(8);
   protected Map<String, List<Transformer>> transformerListCache = new ConcurrentHashMap<>(8);
 
-  private MuleContext muleContext;
+  private final MuleContext muleContext;
```
Comment (plbalbi) [inline core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java]: And this one

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java (hunk 1 above)
Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringXmlConfigurationBuilder.java]: it is being set in the constructor. The `final` is added automatically by the IDE when it detects that  it is not being written anywhere else.

File: core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java (hunk 2 above)
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java]: it is being set in the constructor. The `final` is added automatically by the IDE when it detects that  it is not being written anywhere else.

File: core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java (hunk 3 above)
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/internal/registry/MuleRegistryHelper.java]: it is being set in the constructor. The `final` is added automatically by the IDE when it detects that  it is not being written anywhere else.

**Pull request merged**